spyder_unittest/tests/test_unittestplugin.py

```diff
@@ -1,128 +1,165 @@
+    # Filter out seperators (indicated by action is None) and convert to text
+    menu_items = [action.text() for action in run_menu.get_actions() if action]
 
-@pytest.fixture
-def plugin(qtbot):
-    """Set up the unittest plugin."""
-    res = UnitTestPlugin(None, None)
-    res._main = MagicMock()
-    res._main.get_spyder_pythonpath = MagicMock(return_value='fakepythonpath')
-    res.initialize()
-    return res
+    assert 'Run unit tests' in menu_items
 
 
-@pytest.mark.skip('not clear how to test interactions between plugins')
-def test_plugin_initialization(plugin):
-    assert len(plugin.main.run_menu_actions) == 2
-    assert plugin.main.run_menu_actions[1].text() == 'Run unit tests'
+def test_pythonpath_change(main_window):
+    """
+    Test that pythonpath changes in Spyder propagate to UnitTestWidget.
+    """
+    ppm = main_window.get_plugin(Plugins.PythonpathManager)
+    unittest_plugin = main_window.get_plugin(UnitTestPlugin.NAME)
 
+    new_path = '/some/path'
+    new_path_dict = OrderedDict([(new_path, True)])
+    ppm.get_container()._update_python_path(new_path_dict)
 
-def test_plugin_pythonpath(plugin):
-    # Test signal/slot connection
-    plugin.get_main().sig_pythonpath_changed.connect.assert_called_with(
-        plugin.update_pythonpath)
+    assert unittest_plugin.get_widget().pythonpath == [new_path]
 
-    # Test pythonpath is set to path provided by Spyder
-    assert plugin.get_widget().pythonpath == 'fakepythonpath'
 
-    # Test that change in path propagates
-    plugin.get_main().get_spyder_pythonpath = MagicMock(
-        return_value='anotherpath')
-    plugin.update_pythonpath()
-    assert plugin.get_widget().pythonpath == 'anotherpath'
+def test_default_working_dir(main_window, tmpdir):
+    """
+    Test that plugin's default working dir is current working directory.
+    After creating a project, the plugin's default working dir should be the
+    same as the project directory. When the project is closed again, the 
+    plugin's default working dir should revert back to the current working
+    directory.
+    """
+    projects = main_window.get_plugin(Plugins.Projects)
+    unittest_plugin = main_window.get_plugin(UnitTestPlugin.NAME)
+    project_dir = str(tmpdir)
 
+    assert unittest_plugin.get_widget().default_wdir == os.getcwd()
 
-@pytest.mark.skip('not clear how to test interactions between plugins')
-def test_plugin_wdir(plugin, monkeypatch, tmpdir):
-    # Test signal/slot connections
-    plugin.main.workingdirectory.sig_current_directory_changed.connect.assert_called_with(
-        plugin.update_default_wdir)
-    plugin.main.projects.sig_project_created.connect.assert_called_with(
-        plugin.handle_project_change)
-    plugin.main.projects.sig_project_loaded.connect.assert_called_with(
-        plugin.handle_project_change)
-    plugin.main.projects.sig_project_closed.connect.assert_called_with(
-        plugin.handle_project_change)
+    projects._create_project(project_dir)
```

Comment:
Should we make `_create_project` a public method in the Projects plugin API? It looks like the appropriate method to call to easily create projects.

Reply:
@ccordoba12 Yes, it makes sense to make `_create_project` public. This does commit us to some API stability which may hinder development of projects in the near future; you should have a decent guess whether this is likely to be a problem. The name is also quite similar to the `create_new_project` method but I don't have a solution for that.

Reply:
Ok, thanks for the feedback.